x/stakeibc/keeper/icqcallbacks_delegator_shares.go

```diff
@@ -26,108 +25,99 @@ import (
 
 	// Confirm the validator has actually been slashed
-	if validatorTokens.Equal(validator.DelegationAmt) {
-		k.Logger(ctx).Info(fmt.Sprintf("DelegationCallback: Validator (%s) was not slashed", validator.Address))
+	if delegatedTokens.Equal(validator.DelegationAmt) {
+		k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(chainId, ICQCallbackID_Delegation, "Validator was not slashed"))
 		return nil
-	} else if validatorTokens.GT(validator.DelegationAmt) {
-		errMsg := fmt.Sprintf("DelegationCallback: Validator (%s) tokens returned from query is greater than the DelegationAmt", validator.Address)
-		k.Logger(ctx).Error(errMsg)
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, errMsg)
+	}
+	if delegatedTokens.GT(validator.DelegationAmt) {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Validator (%s) tokens returned from query is greater than the DelegationAmt", validator.Address)
 	}
 
 	// TODO(TESTS-171) add some safety checks here (e.g. we could query the slashing module to confirm the decr in tokens was due to slash)
 	// update our records of the total stakedbal and of the validator's delegation amt
 	// NOTE:  we assume any decrease in delegation amt that's not tracked via records is a slash
 
 	// Get slash percentage
-	slashAmount := validator.DelegationAmt.Sub(validatorTokens)
-
-	weight, err := cast.ToInt64E(validator.Weight)
-	if err != nil {
-		errMsg := fmt.Sprintf("unable to convert validator weight to int64, err: %s", err.Error())
-		k.Logger(ctx).Error(errMsg)
-		return sdkerrors.Wrapf(types.ErrIntCast, errMsg)
-	}
-
+	slashAmount := validator.DelegationAmt.Sub(delegatedTokens)
 	slashPct := sdk.NewDecFromInt(slashAmount).Quo(sdk.NewDecFromInt(validator.DelegationAmt))
-	k.Logger(ctx).Info(fmt.Sprintf("ICQ'd Delegation Amount Mismatch, HostZone: %s, Validator: %s, Delegator: %s, Records Tokens: %v, Tokens from ICQ %v, Slash Amount: %v, Slash Pct: %v!",
-		hostZone.ChainId, validator.Address, queriedDelgation.DelegatorAddress, validator.DelegationAmt, validatorTokens, slashAmount, slashPct))
+	k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(chainId, ICQCallbackID_Delegation,
+		"Validator was slashed! Validator: %s, Delegator: %s, Delegation in State: %v, Delegation from ICQ %v, Slash Amount: %v, Slash Pct: %v",
+		validator.Address, queriedDelgation.DelegatorAddress, validator.DelegationAmt, delegatedTokens, slashAmount, slashPct))
 
 	// Abort if the slash was greater than 10%
 	tenPercent := sdk.NewDec(10).Quo(sdk.NewDec(100))
 	if slashPct.GT(tenPercent) {
-		errMsg := fmt.Sprintf("DelegationCallback: Validator (%s) slashed but ABORTING update, slash is greater than 0.10 (%d)", validator.Address, slashPct)
-		k.Logger(ctx).Error(errMsg)
-		return sdkerrors.Wrapf(types.ErrSlashGtTenPct, errMsg)
+		return sdkerrors.Wrapf(types.ErrSlashGtTenPct, "Validator slashed but ABORTING update, slash is greater than 0.10 (%v)", slashPct)
+	}
+
+	// Update the validator weight and delegation reflect to reflect the slash
```

Comment:
Why is this block now moved down below the slashPct calc?

Reply:
It just seemed to make more logical sense there from a readability perspective. 

The flow was
```go
// get slash amount
// cast weight 
// do something with slash amount
// do another thing with slash amount
// do something with the casted weight
```

seemed like it made more sense to order it/ group as:
```go
// get slash amount
// do something with slash amount
// do another thing with slash amount

// cast weight 
// do something with the casted weight
```